src/jail_run.c

```diff
@@ -313,7 +313,7 @@ jail_run(
         }
     }
 
-    setfsuid(creds->euid);
+    setfsuid(creds->ruid);
     setfsgid(egid);
```

Comment:
Since I'm unfamiliar with these setfsxxx() functions, do they actually have any effect here?

Manpage says "Normally, the value of the filesystem user ID will shadow the value of the effective user ID.  In fact, whenever the effective user ID is changed, the filesystem user ID will also be changed to  the new value of the effective user ID." -> there are setresuid() and setresgid() calls right after setfsxxx() calls -> ???

And "are usually used only by programs such as the Linux NFS server that need to change what user and group ID is used for file access without a corresponding change in the real and effective user and group IDs." -> we do not have such intent either, or?

If the setfsxxx() calls serve no purpose / accomplish nothing, I'd rather remove them altogether.

Reply:
If setfsuid() is done before seteuid(), then there will be no effect. If it is made alone, it will have effects.

Reply:
Hmm. This is good question, I think these are not be effective here because we set effective ids later and apparently changing those changes also file system ids. So basically this euid -> ruid change was ineffective (well, it was never a problem) but the other change is important because that egid value is used later too.
https://linux.die.net/man/7/credentials

Reply:
Removed these, they do nothing at all.